operate/bridge/providers/native_bridge_provider.py

```diff
@@ -0,0 +1,650 @@
+        extra_data = Web3.keccak(text=bridge_request.id)
+
+        if from_token == ZERO_ADDRESS:
+            return self._l1_standard_bridge_contract.build_bridge_eth_to_tx(
+                ledger_api=from_ledger_api,
+                contract_address=from_bridge,
+                sender=from_address,
+                to=to_address,
+                amount=int(to_amount),
+                min_gas_limit=DEFAULT_BRIDGE_MIN_GAS_LIMIT,
+                extra_data=extra_data,
+            )
+
+        return self._l1_standard_bridge_contract.build_bridge_erc20_to_tx(
+            ledger_api=from_ledger_api,
+            contract_address=from_bridge,
+            sender=from_address,
+            local_token=from_token,
+            remote_token=to_token,
+            to=to_address,
+            amount=int(to_amount),
+            min_gas_limit=DEFAULT_BRIDGE_MIN_GAS_LIMIT,
+            extra_data=extra_data,
+        )
+
+    def find_bridge_finalized_tx(
+        self,
+        from_ledger_api: LedgerApi,
+        to_ledger_api: LedgerApi,
+        bridge_request: BridgeRequest,
+        from_block: BlockIdentifier,
+        to_block: BlockIdentifier,
+    ) -> t.Optional[str]:
+        """Return the transaction hash of the event indicating bridge completion."""
+        from_address = bridge_request.params["from"]["address"]
+        from_token = bridge_request.params["from"]["token"]
+        to_address = bridge_request.params["to"]["address"]
+        to_token = bridge_request.params["to"]["token"]
+        to_amount = bridge_request.params["to"]["amount"]
+        to_bridge = self.to_bridge
+        extra_data = Web3.keccak(text=bridge_request.id)
+
+        if from_token == ZERO_ADDRESS:
+            return self._l2_standard_bridge_contract.find_eth_bridge_finalized_tx(
+                ledger_api=to_ledger_api,
+                contract_address=to_bridge,
+                from_=from_address,
+                to=to_address,
+                amount=to_amount,
+                extra_data=extra_data,
+                from_block=from_block,
+                to_block=to_block,
+            )
+
+        return self._l2_standard_bridge_contract.find_erc20_bridge_finalized_tx(
+            ledger_api=to_ledger_api,
+            contract_address=to_bridge,
+            local_token=to_token,
+            remote_token=from_token,
+            from_=from_address,
+            to=to_address,
+            amount=to_amount,
+            extra_data=extra_data,
+            from_block=from_block,
+            to_block=to_block,
+        )
+
+    def get_explorer_link(
+        self, from_ledger_api: LedgerApi, bridge_request: BridgeRequest
+    ) -> t.Optional[str]:
+        """Get the explorer link for a transaction."""
+        if not bridge_request.execution_data:
+            return None
+
+        tx_hash = bridge_request.execution_data.from_tx_hash
+        if not tx_hash:
+            return None
+
+        return f"https://etherscan.io/tx/{tx_hash}"
```

Comment:
we should have these as constants somewhere. That's how they are in frontend also.